aidb/inference/bound_inference_service.py

```diff
@@ -145,12 +145,24 @@ def get_tables(self, columns: List[str]) -> List[str]:
     return list(tables)
 
 
-  async def _insert_in_cache_table(self, row, conn):
-    input_dic = {}
-    for col in self.binding.input_columns:
-      input_dic[self.convert_normalized_col_name_to_cache_col_name(col)] = pandas_dtype_to_native_type(getattr(row, col))
-    insert = self.get_insert()(self._cache_table).values(**input_dic)
-    await conn.execute(insert)
+  async def _insert_in_cache_table(self, inp_rows: List[pd.Series], conn):
+    if len(inp_rows) > 0:
+      inp_rows_df = pd.DataFrame(inp_rows).reset_index(drop=True)
+      inp_rows_df.columns = [self.convert_normalized_col_name_to_cache_col_name(col) for col in inp_rows_df.columns]
+      inp_rows_df = inp_rows_df.astype('object')
```

Comment:
Do we need to cast? Ditto below

Reply:
Yes, we need this. It converts the data to python native types from pandas/numpy types. For e.g. np.int64 to int

Reply:
Can you leave a comment?